clade/proto/sync.proto

```diff
@@ -0,0 +1,43 @@
+syntax = "proto3";
+
+package clade.sync;
+
+import "schema.proto";
+
+message DataSyncCommand {
+  // Path to the Delta Table, relative to the root URL of the store below.
+  // Will be created based on the schema of the RecordBatch if doesn't exist.
+  // e.g. path/to/table
+  string path = 1;
+
+  // Root URL (e.g. s3://my-bucket) and connection options for the table's path
+  schema.StorageLocation store = 2;
+
+  // Primary key columns. Must not be empty: the table must always
+  // have a replica identity / PK.
+  repeated string pk_column = 3;
+
+  // Opaque string identifying data source
+  string origin = 4;
+
+  // Monotonically-increasing transaction number (e.g. the LSN)
+  uint64 sequence_number = 5;
```

Comment:
Just note that it can and does wrap.

Reply:
Thanks, that's a very important point. How often does this occur roughly?

Reply:
LSN does not wrap

Reply:
They shouldn't wrap as per https://www.postgresql.org/docs/16/wal-internals.html and other discussions:

> The numbers do not wrap, but it will take a very, very long time to exhaust the available stock of numbers.

Reply:
My mistake, apologies.